api4/config.go

```diff
@@ -145,6 +145,11 @@ func updateConfig(c *Context, w http.ResponseWriter, r *http.Request) {
 	// Do not allow certificates to be changed through the API
 	cfg.PluginSettings.SignaturePublicKeyFiles = appCfg.PluginSettings.SignaturePublicKeyFiles
 
+	// Do not allow marketplace URL to be toggled through the API if EnableUploads are disabled.
+	if cfg.PluginSettings.EnableUploads != nil && !*cfg.PluginSettings.EnableUploads {
+		cfg.PluginSettings.MarketplaceURL = appCfg.PluginSettings.MarketplaceURL
```

Comment:
We should assign values right?
```suggestion
		*cfg.PluginSettings.MarketplaceURL = *appCfg.PluginSettings.MarketplaceURL
```

Reply:
Yes, the earlier lines also copied the pointer, which is risky. I'll change all of them.